planetterp/static/js/admin-action.js

```diff
@@ -85,8 +85,19 @@ function verifyHelpSuccess(data, args) {
 /* ********* VERIFY PROFESSOR FUNCTIONS ********* */
 function verifyProfessorSuccess(data, args) {
     if (data['form'] != null) {
-        $("#slug-modal-container").html(`${data['form']}`);
-        $(`#slug-modal-${data['id']}`).modal('show');
+        container_id_mappings = {
+            "#info-modal-container": "#info-modal",
+            "#slug-modal-container": "#slug-modal"
+        }
+
+        $(".modal").modal("hide");
+        $(data["success_msg"]).html(`${data['form']}`);
+
+        $('#slug-modal').on('show.bs.modal', function (e) {
+            $('#override').val("false");
+        });
```

Comment:
what is this doing? looks very fragile to me

Reply:
As I was writing this I realized I should've done it slightly differently (see 3c1c85c). The below idea stays the same though:

Context: The verify button in the infoModal uses the same form and validation code as the normal verify button. There's a small portion of that code that triggers the infoModal. We don't want to trigger the modal again if it's already open so I added the "override" field to bypass that part in /admin. 

The code in 3c1c85c just sets the override field back to false when the infoModal is closed.

Reply:
to make sure I understand: the functions in `admin.html` call `sendResponse` with some specified data and an action. you get that data from `$('#action_form').serialize()`, so to make sure you send `override` as part of the data, you set override to true in the action form. But then that form gets reused, so you have to set it *back* to false after `sendResponse` fires.

Do you see how backwards that is? Why is modifying the state of the function necessary, as opposed to just passing along `override: true` to the data sent in `sendResponse` if it's present in `form_args`?

Reply:
Was trying to cut corners but decided to give in and implement it properly.

Reply:
better. I don't understand why `action_form` exists in the first place - as far as I can see, its only purpose is so that you can change the values of its fields and then immediately call `serialize` on it to pass to `sendResponse`. That seems pointless when you could just construct the data to pass to `sendResponse` without going through a form middleman.

But perhaps a cleanup for another day.